downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -221,12 +221,21 @@ abstract class VersionControlSystem {
         abstract fun listRemoteTags(): List<String>
 
         /**
-         * Search (symbolic) names of VCS revisions for matches with the given [project] and [version].
+         * Search (symbolic) names of VCS revisions for a match with the given [project] and [version].
          *
-         * @return A matching VCS revision or an empty String if no match is found.
+         * @return The matching VCS revision, never blank.
+         * @throws IOException If no or multiple matching revisions are found.
          */
-        fun guessRevisionName(project: String, version: String) =
-                filterVersionNames(version, listRemoteTags(), project).firstOrNull() ?: ""
+        fun guessRevisionName(project: String, version: String): String {
+            val versionNames = filterVersionNames(version, listRemoteTags(), project)
+            return when {
+                versionNames.size > 1 ->
+                    throw IOException("Multiple matching tags found for version '$version': $versionNames")
+                versionNames.isEmpty() || versionNames.first().isBlank() ->
```

Comment:
Nit: Maybe make this case the first one?

Also, I've just verified that `filterVersionNames()` never returns empty strings, see https://github.com/heremaps/oss-review-toolkit/pull/438, so we should be able to drop the `versionNames.first().isBlank()` case.

Reply:
It had to be in this order to make sure there is only one element, but if it cannot be blank changing the order makes sense.